src/PrestaShopBundle/Security/Attribute/AdminSecurity.php

```diff
@@ -0,0 +1,198 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShopBundle\Security\Attribute;
+
+use Attribute;
+use RuntimeException;
+use Symfony\Component\ExpressionLanguage\Expression;
+
+/**
+ * Attribute based on the IsGranted attribute, adding information for redirection
+ */
+#[Attribute(Attribute::TARGET_CLASS | Attribute::TARGET_METHOD)]
```

Comment:
Do you really need `Attribute::TARGET_CLASS` too?
I doesn't find any AdminSecurity attribute before a class declaration.

Reply:
There aren't any on our classes actually, but maybe modules do. 
I used the same attributes as on the security attribute of sensio https://github.com/sensiolabs/SensioFrameworkExtraBundle/blob/master/src/Configuration/Security.php